frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx

```diff
@@ -12,87 +12,182 @@ import {
-    e.preventDefault();
-    dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: '' });
-  };
-
   const handleFinishButton = (e: React.FormEvent<EventTarget>) => {
     e.preventDefault();
-    if (state.poolStatus === POOL_PROGRESS.CREATED) {
-      onPoolCreation(state.poolName);
-    }
+    if (state.poolStatus === POOL_PROGRESS.CREATED) onPoolCreation(state.poolName);
     close();
   };
 
-  return state.poolStatus ? (
-    <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
-      {state.poolStatus === POOL_PROGRESS.FAILED && (
-        <Button
-          type="button"
-          variant="secondary"
-          data-test="modal-try-again-action"
-          onClick={handleTryAgainButton}
-        >
-          {t('ceph-storage-plugin~Try Again')}
-        </Button>
-      )}
-      <Button
-        type="submit"
-        variant="primary"
-        data-test="confirm-action"
-        isDisabled={state.poolStatus === POOL_PROGRESS.PROGRESS}
-        id="confirm-action"
-        onClick={handleFinishButton}
-      >
-        {closeLabel}
-      </Button>
-    </ActionGroup>
-  ) : (
+  const modalFooterButtonFactory: ModalFooterButtonFactory = {
+    [POOL_PROGRESS.FAILED]: [
+      {
+        id: 'modal-try-again-action',
+        label: t('ceph-storage-plugin~Try Again'),
+        type: ButtonType.button,
+        variant: ButtonVariant.secondary,
+        onClick: () => dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: '' }),
+      },
+      {
+        id: 'modal-finish-action',
+        label: t('ceph-storage-plugin~Finish'),
+        type: ButtonType.submit,
+        variant: ButtonVariant.primary,
+        onClick: handleFinishButton,
+      },
+    ],
+    [POOL_PROGRESS.NOTALLOWED]: [
+      {
+        id: 'modal-close-action',
+        label: t('ceph-storage-plugin~Close'),
+        type: ButtonType.submit,
+        variant: ButtonVariant.primary,
+        onClick: handleFinishButton,
+      },
+    ],
+    [POOL_PROGRESS.BOUNDED]: [
+      {
+        id: 'modal-close-action',
+        label: t('ceph-storage-plugin~Close'),
+        type: ButtonType.button,
+        variant: ButtonVariant.secondary,
+        onClick: handleFinishButton,
+      },
+      {
+        id: 'modal-go-to-pvc-list-action',
+        label: t('ceph-storage-plugin~Go To Pvc List'),
+        type: ButtonType.submit,
+        variant: ButtonVariant.primary,
+        onClick: () => {
+          redirectToPvcList();
+          close();
+        },
+      },
+    ],
+    [POOL_PROGRESS.CREATED]: [
+      {
+        id: 'modal-finish-action',
+        label: t('ceph-storage-plugin~Finish'),
+        type: ButtonType.submit,
+        variant: ButtonVariant.primary,
+        disable: state.poolStatus === POOL_PROGRESS.PROGRESS,
+        onClick: handleFinishButton,
+      },
+    ],
+    [POOL_PROGRESS.NOTREADY]: [
+      {
+        id: 'modal-finish-action',
```

Comment:
few states have similar prop but i don't want to create a new function to pass the prop, I feel let it be separate, so we can do changes in future easily